src/utilities.jl

```diff
@@ -145,3 +145,96 @@ end
 
 check_positive_definite(M::AbstractMatrix; kwargs...) =
   check_positive_definite(LinearOperator(M); kwargs...)
+
+
+  """
+  solve_shifted_system!(op::LBFGSOperator{T,I,F1,F2,F3}, 
+                        z::AbstractVector{T}, 
+                        σ::T, 
+                        γ_inv::T, 
+                        inv_Cz::AbstractVector{T}, 
+                        p::AbstractVector{T}, 
+                        v::AbstractVector{T}, 
+                        u::AbstractVector{T}) where {T,I,F1,F2,F3}
+
+Computes the regularized L-BFGS step by solving the linear system:
+
+  (B_k + σ_k * I) s = -∇f(x_k)
+
+where `B_k` is the L-BFGS approximation of the Hessian, `σ_k` is a regularization parameter, and `∇f(x_k)` is the gradient at `x_k`.
+
+### Parameters
+- `op::LBFGSOperator{T,I,F1,F2,F3}`: The L-BFGS operator `B_k`. Encodes the curvature information used to approximate the Hessian.
+- `z::AbstractVector{T}`: The vector representing `-∇f(x_k)` (negative gradient at the current iterate).
+- `σ::T`: The regularization parameter, used to shift the Hessian approximation `B_k` by adding a multiple of the identity matrix.
+- `γ_inv::T`: The inverse of the initial curvature `γ_0`, used to initialize the L-BFGS matrix.
```

Comment:
This value is stored inside the operator: https://github.com/JuliaSmoothOptimizers/LinearOperators.jl/blob/main/src/lbfgs.jl#L220